# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsPageContentFrame.cpp
# Commit: ae6b08b3f4de
# Full Hash: ae6b08b3f4def1cf7cf12b8cfa53d0edaad99f97
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2022-02-23 09:38:00
# Regressor Bug: 1714584
# File Overlap Count: 1
# Description:
#   Bug 1714584 - Part 2: Remove nsDisplayList::RemoveBottom() r=mstange
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138153
# ==============================================================================

diff -r a261bb633790 -r ae6b08b3f4de layout/generic/nsPageContentFrame.cpp
--- a/layout/generic/nsPageContentFrame.cpp	Tue Feb 22 23:42:18 2022 +0000
+++ b/layout/generic/nsPageContentFrame.cpp	Tue Feb 22 23:42:18 2022 +0000
@@ -267,10 +267,7 @@
 static void PruneDisplayListForExtraPage(nsDisplayListBuilder* aBuilder,
                                          nsPageFrame* aPage,
                                          nsDisplayList* aList) {
-  nsDisplayList newList;
-
-  while (true) {
-    nsDisplayItem* i = aList->RemoveBottom();
+  for (nsDisplayItem* i : aList->TakeItems()) {
     if (!i) break;
     nsDisplayList* subList = i->GetSameCoordinateSystemChildren();
     if (subList) {
@@ -285,9 +282,8 @@
         continue;
       }
     }
-    newList.AppendToTop(i);
+    aList->AppendToTop(i);
   }
-  aList->AppendToTop(&newList);
 }
 
 static void BuildDisplayListForExtraPage(nsDisplayListBuilder* aBuilder,